tests/unit_tests/test_run_path_creation.py

```diff
@@ -194,6 +194,33 @@ def test_run_template_replace_in_ecl_data_file(key, expected, prior_ensemble):
     ) == f"I WANT TO REPLACE:{expected}"
 
 
+def test_that_error_is_raised_when_data_file_is_badly_encoded(prior_ensemble):
+    config_text = dedent(
+        """
+    NUM_REALIZATIONS 1
+    ECLBASE ECL_CASE<IENS>
+    DATA_FILE MY_DATA_FILE.DATA
+    """
+    )
+    Path("MY_DATA_FILE.DATA").write_text("I WANT TO REPLACE:<DATE>", encoding="utf-8")
```

Comment:
This does not happen in a tmpdir, so puts file in the source directory.

Reply:
Nice catch. Fixed.